### ISSUE TITLE
Documentation request on TraceQL

### ISSUE THREAD

**Issue by cyrille-leclerc** [grafana/tempo#2209]
URL: https://grafana.com/docs/tempo/latest/traceql/

Can we optimize this documentation page on OpenTelemetry traces and thus on OpenTelemetry Semantic Conventions?

Can we consider documenting how to search traces by:

* `service.name`, `service.namespace`, `deployment.environment`
  * {resource.service.name="frontend"}
  * {.service.name="frontend" && .service.namespace="com-shoppingcart"}
  * {.service.name="frontend" && .service.namespace="com-shoppingcart" && .deployment.environment="production"}
* `service.name` and span name
   * {.service.name="frontend" && name=~"/api/orders"}
   * FYI https://github.com/grafana/tempo/issues/2208

Can we also verify the following examples that don't relate to OpenTelemetry Semantic Conventions:

* `{ resource.namespace = "prod" }` and `{ .namespace = "prod" }`: do we mean `resource.service.namespace`?
* `{ resource.region = "eu-west-0" }`: do we mean [`resource.cloud.region`](https://opentelemetry.io/docs/reference/specification/resource/semantic_conventions/cloud/) .

**Comment by joe-elliott** [comment]
@knylander-grafana i agree with this. the current examples are mainly based on our own internal attribute names. Using names consistent with OTel semantic conventions would be helpful.

> {.service.name="frontend" && .service.namespace="com-shoppingcart"}
we are preferring to include the scopes in examples b/c it's significantly faster to search for:
`{resource.service.name="frontend" && resource.service.namespace="com-shoppingcart"}`

> { resource.namespace = "prod" } and { .namespace = "prod" }: do we mean resource.service.namespace?
These would search search for an attributed named "namespace" only. They would not find "service.namespace".

> { resource.region = "eu-west-0" }: do we mean [resource.cloud.region](https://opentelemetry.io/docs/reference/specification/resource/semantic_conventions/cloud/) .

Same. This is searching specifically for an attribute named "region" on trace resources. To search for the linked attribute you would write something like: `{ resource.cloud.region = "eu-west-0" }`

**Comment by cyrille-leclerc** [comment]
@joe-elliott i didn't want to interrupt you but as you already looked at this, I'm happy to work with @knylander-grafana on this

**Comment by knylander-grafana** [comment]
Happy to help with this!

**Comment by cyrille-leclerc** [comment]
FYI improvements in progress on 
* https://github.com/grafana/tempo/pull/2242

**Comment by knylander-grafana** [comment]
Doc updates merged. @cyrille-leclerc Do you think the updates we did addressed the items in this issue? 

Also, the updates to TraceQL doc are currently slated to go live with Tempo 2.1. I can make them live now if I backport the changes. Just let me know if you want them backported.

**Comment by cyrille-leclerc** [comment]
@knylander-grafana I added this last enhancement to the docs https://github.com/grafana/tempo/pull/2271
